vrp/client/map.lua

```diff
@@ -121,14 +122,34 @@ function tvRP.removeNamedMarker(name)
   end
 end
 
+-- markers sort loop
+Citizen.CreateThread(function()
+  while true do
+    Citizen.Wait(1000)
+
+    local px,py,pz = tvRP.getPosition()
+
+    for k,v in pairs(markers) do
+      -- 150 is the min distance wich the markers will be starting to be drawn
+      if GetDistanceBetweenCoords(v.x,v.y,v.z,px,py,pz,true) <= 150.0 then
+        drawing_markers[k] = v
+      else
+        if drawing_markers[k] then drawing_markers[k] = nil end
+      end
+    end
+  end
+end)
+
 -- markers draw loop
 Citizen.CreateThread(function()
   while true do
     Citizen.Wait(0)
 
     local px,py,pz = tvRP.getPosition()
 
-    for k,v in pairs(markers) do
+    -- if this loop get filled with too many markers, the clientside
+    -- starts lagging
+    for k,v in pairs(drawing_markers) do
       -- check visibility
       if GetDistanceBetweenCoords(v.x,v.y,v.z,px,py,pz,true) <= v.visible_distance then
```

Comment:
Although you didn't write this it would be nice to do the maths on this also.

Reply:
Rather than `GetDistanceBetweenCoords`

Reply:
nice idea. it was just a quick fix for markers draw. I can edit this and commit on my fork if you want